internal/storage/bucket_handle_test.go

```diff
@@ -765,3 +765,186 @@ func (t *BucketHandleTest) TestNameMethod() {
 
 	AssertEq(TestBucketName, name)
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWhenDstObjectDoesNotExist() {
+	var notfound *gcs.NotFoundError
+	_, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: dstObjectName,
+		})
+	AssertTrue(errors.As(err, &notfound))
+	srcObj1, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+	AssertEq(nil, err)
+	AssertNe(nil, srcObj1)
+	srcObj2, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestSubObjectName,
+		})
+	AssertEq(nil, err)
+	AssertNe(nil, srcObj2)
+
+	// Add DstGenerationPrecondition = 0 as the Destination object doesn't exist.
+	// Note: fake-gcs-server doesn't respect precondition checks but still adding
+	// to make sure that it works when precondition checks are supported or we
+	// shift to different testing storage.
+	var zeroPreCond int64 = 0
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       dstObjectName,
+			DstGenerationPrecondition:     &zeroPreCond,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestObjectName,
+				},
+				{
+					Name: TestSubObjectName,
+				},
+			},
+			ContentType: ContentType,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			ContentLanguage:    ContentLanguage,
+			ContentEncoding:    ContentEncoding,
+			CacheControl:       CacheControl,
+			ContentDisposition: ContentDisposition,
+			CustomTime:         CustomTime,
+			EventBasedHold:     true,
+			StorageClass:       StorageClass,
+			Acl:                nil,
+		})
+
+	AssertEq(nil, err)
+	// Validation of srcObject1 to ensure that it is not effected.
```

Comment:
contents are compared in the below test but not here.

Reply:
Added comparison here as well.